zookeeper-server/src/main/java/org/apache/zookeeper/server/ZKDatabase.java

```diff
@@ -120,6 +124,23 @@ public ZKDatabase(FileTxnSnapLog snapLog) {
             snapshotSizeFactor = DEFAULT_SNAPSHOT_SIZE_FACTOR;
         }
         LOG.info("{} = {}", SNAPSHOT_SIZE_FACTOR, snapshotSizeFactor);
+
+        try {
+            commitLogCount = Integer.parseInt(
+                    System.getProperty(COMMIT_LOG_COUNT,
+                            Integer.toString(DEFAULT_COMMIT_LOG_COUNT)));
+            if (commitLogCount < DEFAULT_COMMIT_LOG_COUNT) {
+                commitLogCount = DEFAULT_COMMIT_LOG_COUNT;
+                LOG.warn("The configured {} is less than the recommended {}" +
+                                ", going to use the recommended one",
+                        COMMIT_LOG_COUNT, DEFAULT_COMMIT_LOG_COUNT);
+            }
+        } catch (NumberFormatException e) {
+            LOG.error("Error parsing {} - use default value {}",
+                    COMMIT_LOG_COUNT, DEFAULT_COMMIT_LOG_COUNT);
+            commitLogCount = DEFAULT_COMMIT_LOG_COUNT;
+        }
+        LOG.info(COMMIT_LOG_COUNT + "=" + commitLogCount);
```

Comment:
nit about log format: `LOG.info("{}={}", COMMIT_LOG_COUNT, commitLogCount);` (thanks to @enixon who taught me this recently).